pkg/git/git.go

```diff
@@ -0,0 +1,367 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git
+
+import (
+	"fmt"
+	"net/url"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	GitHubHost    string = "github.com"
+	RawGitHubHost string = "raw.githubusercontent.com"
+	GitLabHost    string = "gitlab.com"
+	BitbucketHost string = "bitbucket.org"
+)
+
+type Url struct {
```

Comment:
Do we want to rename this back to GitUrl to be consistent with the comments,  mock type and util functions?  Also, the name `Url` is pretty generic